tools/SourceKit/include/SourceKit/Core/LangSupport.h

```diff
@@ -679,11 +683,11 @@ class LangSupport {
                             ArrayRef<const char *> Args,
                             std::function<void(const RangeInfo&)> Receiver) = 0;
 
-  virtual void
-  getCursorInfoFromUSR(StringRef Filename, StringRef USR,
-                       bool CancelOnSubsequentRequest,
-                       ArrayRef<const char *> Args,
-                     std::function<void(const CursorInfoData &)> Receiver) = 0;
+  virtual void getCursorInfoFromUSR(
+      StringRef Filename, StringRef USR, bool CancelOnSubsequentRequest,
+      ArrayRef<const char *> Args,
+      llvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem> FileSystem,
+      std::function<void(const CursorInfoData &)> Receiver) = 0;
```

Comment:
What do you think about adding the `FileSystem` parameter to the end of the original parameter list?

Reply:
will be addressed by the small refactoring